platforms/jvm/language-groovy/src/main/java/org/gradle/api/tasks/compile/GroovyCompileOptions.java

```diff
@@ -272,8 +272,19 @@ public GroovyForkOptions getForkOptions() {
     /**
      * Sets options for running the Groovy compiler in a separate process. These options only take effect
      * if {@code fork} is set to {@code true}.
+     *
+     * @deprecated Setting the value of certain {@link Nested @Nested} properties is deprecated and will be removed in Gradle 9.0.
+     * Inner {@code Property}-based properties need to have consistent references that can be used as the source for other lazy APIs.
+     * Set the individual properties of the nested object instead.
```

Comment:
This applies to the other methods too.

```suggestion
     * @deprecated Setting a new instance of this property is unnecessary. This method will be removed in Gradle 9.0. Use {@link #forkOptions(Action)} instead.
```